### PR TITLE
Implemented FetchBill basic functionality

### PR DESCRIPTION
**Pull request by hmtey**
Only works with guests without bills.
Unable to proceed further without proper billing system in place.

### CODE REVIEW

**Comment by hmtey** [comment]
Trying to figure out how personbuilder works

File: docs/UserGuide.adoc [hunk 1]
```diff
@@ -34,8 +34,8 @@ image::Ui.png[width="790"]
 e.g. typing *`help`* and pressing kbd:[Enter] will open the help window.
 .  Some example commands you can try:
 
-* **`checkin`**`n/ Smith r/101` : checks in `Smith` to room `101` in the hotel.
-* **`checkout`**`r/101` : checks out guest from room `101` and removes guest from guest list.
+* **`checkin`**`n/ Smith rn/101` : checks in `Smith` to room `101` in the hotel.
```
Comment (dawidjaja) [inline docs/UserGuide.adoc]: Could you help change this according to the current format?

File: docs/UserGuide.adoc [hunk 2]
```diff
@@ -63,35 +63,33 @@ Format: `checkin i/ID rn/ROOM_NUMBER td/TO_DATE`
...
+* `checkout n/Smith rn/101`
 
 === Shows a list of all the guests in the hotel. +
 Format: `list`
 
 === Fetching the bill of a guest : `bill`
 
 Retrieves the bill of a guest. +
-Format: `bill n/NAME [r/ROOM_NUMBER]...`
+Format: `bill i/ID`
```
Comment (dawidjaja) [inline docs/UserGuide.adoc]: The format in your code has an optional room number. Please make it consistent.

File: src/main/java/seedu/address/logic/commands/FetchBillCommand.java [hunk 3]
```diff
@@ -13,17 +20,68 @@
...
+        requireNonNull(model);
+
+        Optional<Person> person = model.findPersonWithId(personId);
+
+        if (person.isEmpty()) {
+            throw new CommandException(MESSAGE_GUEST_NONEXISTENT);
+        }
+
+        if (isSpecific) {
+            Optional<Room> room = model.findRoom(personId.toString());
```
Comment (dawidjaja) [inline src/main/java/seedu/address/logic/commands/FetchBillCommand.java]: model.findRoom takes roomNum as parameters. We haven't made findRoomByPersonId() for now. Could you either create the method in the model manager or change this part?

File: src/main/java/seedu/address/model/ModelManager.java [hunk 4]
```diff
@@ -218,6 +220,20 @@ public void bookRoom(Booking booking) {
...
+    public void fetchBillList(Person person) {
+        String[] dummy = new String[] {"bnfajdnnlasdf"};
+        updateFilteredPersonList(new NameContainsKeywordsPredicate(Arrays.asList(dummy)));
+    }
+
+    @Override
+    public void fetchBill(Person person, String roomNum) {
+        String[] dummy = new String[]{"bnfajdnnlasdf"};
+        updateFilteredPersonList(new NameContainsKeywordsPredicate(Arrays.asList(dummy)));
+    }
```
Comment (dawidjaja) [inline src/main/java/seedu/address/model/ModelManager.java]: What does dummy here do?

File: src/main/java/seedu/address/model/ModelManager.java [hunk 5]
```diff
@@ -218,6 +220,20 @@ public void bookRoom(Booking booking) {
         bookingList.add(booking);
     }
 
+
+    // to update accordingly when implementing billing system.
```
Comment (dawidjaja) [inline src/main/java/seedu/address/model/ModelManager.java:222]: We should standardize and use /** */

File: src/test/java/seedu/address/logic/parser/FetchBillCommandParserTest.java [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+
+public class FetchBillCommandParserTest {
+    private static final Person Guest = new PersonBuilder().build();
+    private static final PersonId PersonId = Guest.getPersonId();
+    private static final Room Room = new RoomBuilder().build();
+    private static final String RoomNum = Room.getRoomNum();
+
+    private FetchBillCommandParser parser = new FetchBillCommandParser();
+    /*
+    @Test
```
Comment (dawidjaja) [inline src/test/java/seedu/address/logic/parser/FetchBillCommandParserTest.java:26]: Why are these commented out?

**Review by dawidjaja — CHANGES_REQUESTED** [review]
Could you read these comments?

File: src/main/java/seedu/address/model/ModelManager.java (hunk 4 above)
Comment (hmtey) [reply src/main/java/seedu/address/model/ModelManager.java]: It produces an empty view for guests without bills (for now), will need to change these two methods once the bill system is implemented

File: src/test/java/seedu/address/logic/parser/FetchBillCommandParserTest.java (hunk 6 above)
Comment (hmtey) [reply src/test/java/seedu/address/logic/parser/FetchBillCommandParserTest.java:26]: They cause checkstyle errors due to personbuilder, thus as mentioned above I need to figure this out. ReserveCommandParserTest seems to have the same issue too.

File: src/main/java/seedu/address/model/ModelManager.java (hunk 5 above)
Comment (youaremysky99) [reply src/main/java/seedu/address/model/ModelManager.java:222]: I think /** */ is just used for docs. For normal comments, it's better to use //

Comment (hmtey) [reply src/main/java/seedu/address/model/ModelManager.java:222]: I left it as //, is that alright?

**Review by dawidjaja — APPROVED** [review]
LGTM

**Pull request merged**